### PR TITLE
Refactor xpack

### PR DESCRIPTION
**Pull request by philkra**
as titled. ref: #409

### CODE REVIEW

File: specification/xpack/usage/types.ts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+export class AlertingExecution {
+  actions: Dictionary<string, ExecutionAction>
+}
+
+export class AlertingInput {
+  input: Dictionary<string, UsageCount>
+  trigger: Dictionary<string, UsageCount>
+}
+
+export class AnalyticsStatsUsage {
```
Comment (delvedor) [inline specification/xpack/usage/types.ts:45]: There is no need to name all of this types `*Usage`, as those are already defined inside the `usage` folder.
You should either drop the `Usage` from the name, or move all of these definitions to the `xpack/_types` folder.

**Review by delvedor — CHANGES_REQUESTED** [review]

File: specification/xpack/usage/types.ts (hunk 1 above)
Comment (philkra) [reply specification/xpack/usage/types.ts:45]: Right! Will rename

File: specification/xpack/usage/types.ts [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+ */
+
+import { Phases } from '@ilm/_types/Phases'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { ByteSize, EmptyObject, Field, Name } from '@_types/common'
+import { Job, JobStatistics } from '@_types/ml/Job'
+import { integer, long, uint, ulong } from '@_types/Numeric'
+import { Statistics } from '@_types/slm/Statistics'
+
+export class Usage {
```
Comment (delvedor) [inline specification/xpack/usage/types.ts:27]: Let's rename this type to `Base`, so the type name will be `XPack.Usage.Base` :)

```suggestion
export class Base {
```

Comment (philkra) [reply specification/xpack/usage/types.ts]: done

**Comment by philkra** [comment]
addressed feedback.

**Review by delvedor — APPROVED** [review]
LGTM once the conflicts are resolved.

**Comment by philkra** [comment]
@delvedor - conflicts resolved

**Review by delvedor — APPROVED** [review]

**Pull request merged**